core/src/main/java/org/jdbi/v3/core/result/ResultIterable.java

```diff
@@ -81,6 +82,40 @@
     @Override
     ResultIterator<T> iterator();
 
+    /**
+     * Returns a {@code ResultIterable<U>} derived from this {@code ResultIterable<T>}, by
+     * transforming elements using the given mapper function.
+     *
+     * @param mapper function to apply to elements of this ResultIterable
+     * @param <U>    Element type of the returned ResultIterable
+     * @return the new ResultIterable
+     */
+    default <U> ResultIterable<U> map(Function<T, U> mapper) {
```

Comment:
Would `Function<? super T, ? extends U>` be more appropriate? This is convenient for passing `Function`'s, or specifically method references, which are compatible with the `T` input and `U` output but are not strictly those types, no casting needed ([Stream#map](https://github.com/JetBrains/jdk8u_jdk/blob/master/src/share/classes/java/util/stream/Stream.java#L184)).

Reply:
Done